applications/callflow/src/cf_endpoint.erl

```diff
@@ -556,9 +556,13 @@ should_create_endpoint([Routine|Routines], Endpoint, Properties, Call) when is_f
 maybe_owner_called_self(Endpoint, Properties, Call) ->
     maybe_owner_called_self(Endpoint, Properties, whapps_call:resource_type(Call), Call).
 
--spec maybe_owner_called_self(wh_json:object(), wh_json:object(),  binary(), whapps_call:call()) ->
+-spec maybe_owner_called_self(wh_json:object(), wh_json:object(),  api_binary(), whapps_call:call()) ->
                                      'ok' |
                                      {'error', 'owner_called_self'}.
+maybe_owner_called_self(_, _, 'undefined', _) ->
+    lager:error("whapps_call resource type is undefined"),
+    wh_util:log_stacktrace(),
+    'ok';
```

Comment:
Shouldn't an error tuple be returned or exception thrown? Maybe `throw({'error', 'no_resource_type'});` ?